wallet/wallet.h

```diff
@@ -89,7 +89,7 @@ struct wallet_payment {
 	struct sha256 payment_hash;
 	enum wallet_payment_status status;
 	struct pubkey *destination;
-	u64 msatoshi;
+	u64 *msatoshi;
```

Comment:
I think this whole changeset is unnecessary.  Payments we're making must have a known value, AFAICT.  The other two patches stand alone.

Reply:
The problem is that a payment entry is created when an invoice is created.  (This seems strange to me. I think it makes more sense to create a payment entry when the invoice is actually paid).

In the case that the invoice is "any", what value should we put in the msatoshi field of the payment entry created while creating the invoice?

Reply:
Ah, pretty sure we should create it in response to an actual incoming payment. @cdecker?

Using the same type for outgoing and incoming payment is pretty ugly; as is the layering between the wallet and the rest of the code here.   I'd rather have the wallet code take over struct invoice and struct pay_command, but that's a bigger change.

Reply:
I could remove this changeset, and instead just load `payment.msatoshi = 0;` for any-amount invoices for now. Then we can open a separate issue for properly creating incoming payments when actual incoming payment occurs.

Reply:
The idea of this was to be a backing for the listing of payments, so we don't have to merge invoices, incoming payments and outgoing payments. It'd be rather unexpected for people to issue an invoice only to not see it in their list.

But I'm happy to defer the creation until the payment comes in, and not show invoiced payments in the transfer list.

Reply:
@cdecker I should have reviewed this harder: it's horribly confusing.  There's no payment yet!  You don't want this is a "listpayments" API...

From a user POV, outgoing and incoming payments are completely different creatures.  I'm not even sure why you'd want to see incoming payments at all, in fact.  AFAICT they're *always* immediately paid once we pay invoice.  So, invoices tell you what's paid.